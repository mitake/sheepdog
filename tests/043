#!/bin/bash

# Test disk error handling

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

for i in `seq 0 4`; do
    _start_sheep $i
done

_wait_for_sheep 5

$COLLIE cluster format

$COLLIE vdi create test 40M

for i in `seq 0 10`; do
    echo $i | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512
done

for i in 0 1 2 3 4; do
    $COLLIE vdi read test -p 700$i | md5sum
    $COLLIE cluster info -p 700$i | _filter_cluster_info
done

# remove obj directory to occur EIO
rm -rf $STORE/4/obj

for i in `seq 0 10`; do
    echo $(($i + 100)) | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512
done

for i in 0 1 2 3 4; do
    $COLLIE vdi read test -p 700$i | md5sum
    $COLLIE cluster info -p 700$i | _filter_cluster_info
done

# check whether sheep 4 can receive confchg event
_kill_sheep 3

for i in `seq 0 10`; do
    echo $(($i + 200)) | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512
done

for i in 0 1 2 4; do
    $COLLIE vdi read test -p 700$i | md5sum
    $COLLIE cluster info -p 700$i | _filter_cluster_info
done

# check whether sheep 4 can receive notify event
$COLLIE vdi create test2 20M

for i in 0 1 2 4; do
    $COLLIE vdi list -p 700$i | _filter_short_date
done
