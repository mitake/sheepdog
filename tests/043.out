QA output created by 043
using backend farm store
e6fffdc037c9f623d1c03fda7ef9cabf  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
e6fffdc037c9f623d1c03fda7ef9cabf  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
e6fffdc037c9f623d1c03fda7ef9cabf  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
e6fffdc037c9f623d1c03fda7ef9cabf  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
e6fffdc037c9f623d1c03fda7ef9cabf  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
c2f72711cc1e4781841aa416f0b43e0d  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
c2f72711cc1e4781841aa416f0b43e0d  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
c2f72711cc1e4781841aa416f0b43e0d  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
c2f72711cc1e4781841aa416f0b43e0d  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
c2f72711cc1e4781841aa416f0b43e0d  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
99970159bd855e83ced168eca46ae6c5  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      3 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
99970159bd855e83ced168eca46ae6c5  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      3 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
99970159bd855e83ced168eca46ae6c5  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      3 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
99970159bd855e83ced168eca46ae6c5  -
Cluster status: running

Cluster created at DATE

Epoch Time           Version
DATE      3 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003, 127.0.0.1:7004]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         1   40 MB   40 MB  0.0 MB DATE   7c2b25     3              
  test2        1   20 MB  0.0 MB  0.0 MB DATE   fd3815     3              
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         1   40 MB   40 MB  0.0 MB DATE   7c2b25     3              
  test2        1   20 MB  0.0 MB  0.0 MB DATE   fd3815     3              
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         1   40 MB   40 MB  0.0 MB DATE   7c2b25     3              
  test2        1   20 MB  0.0 MB  0.0 MB DATE   fd3815     3              
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         1   40 MB   40 MB  0.0 MB DATE   7c2b25     3              
  test2        1   20 MB  0.0 MB  0.0 MB DATE   fd3815     3              
